src/track/serato/beatgrid.cpp

```diff
@@ -511,10 +513,12 @@ void SeratoBeatGrid::setBeats(BeatsPointer pBeats,
     // Finally, create the terminal marker.
     const double positionSecs =
             streamInfo.getSignalInfo().frames2secs(
-                    streamInfo.getSignalInfo().samples2frames(
-                            currentBeatPositionSamples)) -
+                    static_cast<SINT>(currentBeatPositionFrames)) -
             timingOffsetSecs;
-    const double bpm = pBeats->getBpmAroundPosition(currentBeatPositionSamples, 1);
+    const double bpm = pBeats->getBpmAroundPosition(
```

Comment:
What is this bpm value about? 
Do we have documentations about SeratoBeatGridTerminalMarker? 
If not it would be nice if you can add a comments about it. 
Which Terminal does it refer to?

Reply:
https://github.com/Holzhaus/serato-tags/blob/master/docs/serato_beatgrid.md#beatgrid-markers

Reply:
"terminal = the last one" is not my first assassination, I think to the German "Klemme" or Flight Terminal. 
How about renaming it to "SeratoBeatGridTerminator" or "SeratoBeatGridFinalMarker" or such? 

What is the bpm value? Is it the nominal value for the whole track?

Reply:
Regarding the naming: In German you also say "terminalsymbol" und "nichtterminalsymbol": https://de.m.wikipedia.org/wiki/Regul%C3%A4re_Grammatik#Beschreibung_des_Ableitungsvorgangs

Reply:
> Bei rechtsregulären Grammatiken darf die rechte Seite w 2 {\displaystyle w_{2}} w_{2} einer Produktion w 1 → w 2 {\displaystyle w_{1}\to w_{2}} {\displaystyle w_{1}\to w_{2}} nur das leere Wort, ein oder mehrere Terminalsymbole oder ein oder mehrere Terminale gefolgt von einem einzigen Nichtterminal sein.

??? Ich verstehe da nur Bahnhof

If I am the only person who did not initially understand the purpose I am fin to keep it. 
If you also think the pair:   
SeratoBeatGridMarker / SeratoBeatGridFinalMarker
is more fluent, let's change it.

Reply:
> > Bei rechtsregulären Grammatiken darf die rechte Seite w 2 {\displaystyle w_{2}} w_{2} einer Produktion w 1 → w 2 {\displaystyle w_{1}\to w_{2}} {\displaystyle w_{1}\to w_{2}} nur das leere Wort, ein oder mehrere Terminalsymbole oder ein oder mehrere Terminale gefolgt von einem einzigen Nichtterminal sein.
> 
> ??? Ich verstehe da nur Bahnhof

*This is just an example for a regular grammar, has nothing to do with this PR:*

Regulaere Grammatiken beschreiben mittel Produktionsregeln Wie zeichenketten gebildet werden koennen. Das gegenstueck dazu sind regulare Ausdrucke (RegEx), die das ganze dann Parsen. Wenn ich beispielsweise beschreiben will, wie man woerter bildet die zu der RegEx "(a*b)?" passen, saehe die Grammatik dazu so aus:

Terminalsymbole: T = {a, b}
Nichtterminalsymbole: N = {S, A }
Startsymbol: S
Produktionsregeln: P = {
    S → ε,
    S →A,
    A → aA,
    A →b
}

Das oben ist eine rechtsregulaere Gramatik (wegen A → aA), bei linksregulaeren Gramatiken muesste es die From (A → Aa) haben und die anderen Produktionsregeln entsprechend angepasst werden :D

> SeratoBeatGridMarker / SeratoBeatGridFinalMarker

I think this is a bit confusing because a terminal (final) marker is still a marker. My initial implementation used a single list and had both the nonterminal markers and the terminal markers inherit from a marker type.

Reply:
I am happy with any other name that is more common. It would be interesting how Serato calls them :-)